src/methods/function.js

```diff
@@ -662,7 +663,7 @@ function InternalCall(
   }
 
   // 2. If F's [[FunctionKind]] internal slot is "classConstructor", throw a TypeError exception.
-  if (F.$FunctionKind === "classConstructor")
+  if (F instanceof ECMAScriptFunctionValue && F.$FunctionKind === "classConstructor")
```

Comment:
@NTillmann, I may not agree with this. Moving $FunctionKind to OrdinaryFunctionValue will force it to become optional by adding "void". I do not think this is helpful for flow. I would rather force any places accessing $FunctionKind field to explicitly check for ECMAScriptFunctionValue. This avoids bugs like accidentally use $FunctionKind in NativeFunctionValue code paths.